[CI] EsqlNodeSubclassTests testReplaceChildren {class org.elasticsearch.xpack.esql.plan.logical.TopN} failing

**Build Scans:**
- [elasticsearch-intake #38212 / part3](https://gradle-enterprise.elastic.co/s/3ad7noh3qrif6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests" -Dtests.method="testReplaceChildren {class org.elasticsearch.xpack.esql.plan.logical.TopNBy}" -Dtests.seed=33C54E81D1355A3A -Dtests.locale=eo -Dtests.timezone=US/Alaska -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testReplaceChildren%20%7Bclass%20org.elasticsearch.xpack.esql.plan.logical.TopN%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testReplaceChildren {class org.elasticsearch.xpack.esql.plan.logical.TopN}
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```
Caused by: java.lang.UnsupportedOperationException: returnType() called on unresolved PromQL function [XdwUk]
```
Seems to be the same as https://github.com/elastic/elasticsearch/issues/147232.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -445,21 +445,12 @@ tests:
 - class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
   method: testBulkOperations {useLegacyFormat=false useSyntheticSource=true}
   issue: https://github.com/elastic/elasticsearch/issues/147177
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testReplaceChildren {class org.elasticsearch.xpack.esql.plan.logical.promql.operator.VectorBinaryComparison}
-  issue: https://github.com/elastic/elasticsearch/issues/147195
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testTransform {class org.elasticsearch.xpack.esql.plan.logical.promql.UnresolvedPromqlFunction}
-  issue: https://github.com/elastic/elasticsearch/issues/147199
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=aggregations/terms_text_docvalues/Indexing disabled}
   issue: https://github.com/elastic/elasticsearch/issues/147200
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=index/100_field_name_length_limit/Test field name length limit array}
   issue: https://github.com/elastic/elasticsearch/issues/147205
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testInfoParameters {class org.elasticsearch.xpack.esql.plan.logical.promql.UnresolvedPromqlFunction}
-  issue: https://github.com/elastic/elasticsearch/issues/147223
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:tstep.docs_tstep_by_one_hour_duration_as_string}
   issue: https://github.com/elastic/elasticsearch/issues/147228
@@ -481,36 +472,9 @@ tests:
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:tstep.tstep_with_request_timestamp_bounds_only}
   issue: https://github.com/elastic/elasticsearch/issues/147228
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testReplaceChildren {class org.elasticsearch.xpack.esql.plan.logical.TopNBy}
-  issue: https://github.com/elastic/elasticsearch/issues/147232
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testReplaceChildren {class org.elasticsearch.xpack.esql.plan.logical.TopN}
-  issue: https://github.com/elastic/elasticsearch/issues/147233
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/KNN Vector similarity search only}
   issue: https://github.com/elastic/elasticsearch/issues/147251
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testReplaceChildren {class org.elasticsearch.xpack.esql.plan.logical.promql.UnresolvedPromqlFunction}
-  issue: https://github.com/elastic/elasticsearch/issues/147258
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testTransform {class org.elasticsearch.xpack.esql.plan.physical.SubqueryExec}
-  issue: https://github.com/elastic/elasticsearch/issues/147270
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testTransform {class org.elasticsearch.xpack.esql.plan.physical.OutputExec}
-  issue: https://github.com/elastic/elasticsearch/issues/147271
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testTransform {class org.elasticsearch.xpack.esql.plan.physical.HashJoinExec}
-  issue: https://github.com/elastic/elasticsearch/issues/147272
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testTransform {class org.elasticsearch.xpack.esql.plan.physical.LookupJoinExec}
-  issue: https://github.com/elastic/elasticsearch/issues/147273
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testReplaceChildren {class org.elasticsearch.xpack.esql.plan.logical.TimeSeriesAggregate}
-  issue: https://github.com/elastic/elasticsearch/issues/147280
-- class: org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests
-  method: testReplaceChildren {class org.elasticsearch.xpack.esql.plan.logical.OrderBy}
-  issue: https://github.com/elastic/elasticsearch/issues/147281
 
 # Examples:
 #

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/tree/EsqlNodeSubclassTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/tree/EsqlNodeSubclassTests.java
@@ -653,7 +653,8 @@ public class EsqlNodeSubclassTests<T extends B, B extends Node<B>> extends NodeS
 
     public static <T extends Node<?>> T makeNode(Class<? extends T> nodeClass) throws Exception {
         if (Modifier.isAbstract(nodeClass.getModifiers())) {
-            nodeClass = randomFrom(innerSubclassesOf(nodeClass));
+            var subclasses = innerSubclassesOf(nodeClass);
+            nodeClass = randomValueOtherThanMany(UNRESOLVED_CLASSES::contains, () -> randomFrom(subclasses));
         }
         Class<?> testSubclassFor = testClassFor(nodeClass);
         if (testSubclassFor != null) {

